gateway/ln-gateway/src/gateway_module_v2/mod.rs

```diff
@@ -279,10 +358,12 @@ impl GatewayClientModuleV2 {
             )
             .await?;
 
-        Ok(())
+        self.subscribe_receive(operation_id)
```

Comment:
Is this necessary to have here? Caller can always call this after calling `relay_direct_swap` and decide if they want to wait for a particular state or not.